src/Service.Tests/SqlTests/GraphQLMutationTests/GraphQLMutationTestBase.cs

```diff
@@ -677,6 +677,185 @@ public async Task DeleteMutationWithVariablesAndMappings(string dbQuery, string
             Assert.AreEqual(result.RootElement.GetProperty("count").GetInt64(), 0);
         }
 
+        /// <summary>
+        /// Performs concurrent update mutations on the same item and validates that the responses
+        /// returned are consistent
+        /// gQLMutation1 : Updates the title column of Book table to New Title
+        /// gQLMutation2 : Updates the title column of Book table to Updated Title
+        /// The title field in the responses returned for each of the mutations should be
+        /// the same value it had written to the table.
+        /// </summary>
+        [TestMethod]
+        public async Task TestParallelUpdateMutations()
+        {
+            string graphQLMutationName = "updatebook";
+            string gQLMutation1 = @"
+                mutation {
+                    updatebook(id : 1, item: { title: ""New Title"" })
+                    {
+                        title
+                    }
+                }";
+
+            string gQLMutation2 = @"
+                mutation {
+                    updatebook(id : 1, item: { title: ""Updated Title"" })
+                    {
+                        title
+                    }
+                }";
+
+            Task<JsonElement> responeTask1 = ExecuteGraphQLRequestAsync(gQLMutation1, graphQLMutationName, isAuthenticated: true);
+            Task<JsonElement> responseTask2 = ExecuteGraphQLRequestAsync(gQLMutation2, graphQLMutationName, isAuthenticated: true);
+
+            JsonElement response1 = await responeTask1;
+            JsonElement response2 = await responseTask2;
+
+            Assert.AreEqual("{\"title\":\"New Title\"}", response1.ToString());
+            Assert.AreEqual("{\"title\":\"Updated Title\"}", response2.ToString());
+        }
+
+        /// <summary>
+        /// Performs concurrent insert mutation on a table where the PK is auto-generated.
+        /// Since, PK is auto-generated, essentially both the mutations are operating on
+        /// different items. Therefore, both the mutations should succeed.
+        /// </summary>
+        [TestMethod]
+        public async Task TestParallelInsertMutationPKAutoGenerated()
```

Comment:
Any way to add tests for REST? Because the query builder/transaction/isolation level changes in this PR apply to all requests

Reply:
The behavior described in this issue happens only in the case of graphQL mutations. This does not happen for REST Put/Patch requests. So, there is no way to write an integration test that produces inconsistent results (Both the REST requests seeing the same results).

To err on the side of caution, I have tested concurrent REST mutations using the k6 test suite and Jmeter (to validate if any issues related to transactions shows up when the volume of requests trying to update the same row is high).